coriolis/tests/osmorphing/test_suse.py

```diff
@@ -0,0 +1,376 @@
+        ])
+        mock_test_path_chroot.assert_has_calls([
+            mock.call('/boot/efi/EFI/suse/grub.cfg'),
+            mock.call('/boot/grub2/grub.cfg')
+        ])
+
+    @mock.patch.object(base.BaseLinuxOSMorphingTools, '_exec_cmd_chroot')
+    def test__run_dracut(self, mock_exec_cmd_chroot):
+        self.morphing_tools._run_dracut()
+
+        mock_exec_cmd_chroot.assert_called_once_with(
+            "dracut --regenerate-all -f")
+
+    @mock.patch.object(base.BaseLinuxOSMorphingTools, '_exec_cmd_chroot')
+    def test__run_mkinitrd_success(self, mock_exec_cmd_chroot):
+        self.morphing_tools._run_mkinitrd()
+
+        mock_exec_cmd_chroot.assert_called_once_with(
+            "mkinitrd")
+
+    @mock.patch.object(base.BaseLinuxOSMorphingTools, '_exec_cmd_chroot')
+    def test__run_mkinitrd_with_exception(self, mock_exec_cmd_chroot):
+        mock_exec_cmd_chroot.side_effect = Exception()
+
+        with self.assertLogs('coriolis.osmorphing.suse', level=logging.WARN):
+            self.morphing_tools._run_mkinitrd()
+
+    @mock.patch.object(
+        base.BaseLinuxOSMorphingTools, '_version_supported_util'
+    )
+    @mock.patch.object(suse.BaseSUSEMorphingTools, '_run_mkinitrd')
+    @mock.patch.object(suse.BaseSUSEMorphingTools, '_run_dracut')
+    def test__rebuild_initrds(self, mock_run_dracut, mock_run_mkinitrd,
+                              mock_version_supported_util):
+        mock_version_supported_util.return_value = True
+
+        self.morphing_tools._rebuild_initrds()
+
+        mock_run_mkinitrd.assert_called_once()
+        mock_run_dracut.assert_not_called()
+
+    @mock.patch.object(
+        base.BaseLinuxOSMorphingTools, '_version_supported_util'
+    )
+    @mock.patch.object(suse.BaseSUSEMorphingTools, '_run_mkinitrd')
+    @mock.patch.object(suse.BaseSUSEMorphingTools, '_run_dracut')
+    def test__rebuild_initrds_old_version(self, mock_run_dracut,
+                                          mock_run_mkinitrd,
+                                          mock_version_supported_util):
+        mock_version_supported_util.return_value = False
+
+        self.morphing_tools._rebuild_initrds()
+
+        mock_run_mkinitrd.assert_not_called()
+        mock_run_dracut.assert_called_once()
+
+    @mock.patch.object(base.BaseLinuxOSMorphingTools, '_exec_cmd_chroot')
+    def test__has_systemd(self, mock_exec_cmd_chroot):
+        result = self.morphing_tools._has_systemd()
+
+        self.assertTrue(result)
+        mock_exec_cmd_chroot.assert_called_once_with("rpm -q systemd")
+
+    @mock.patch.object(base.BaseLinuxOSMorphingTools, '_exec_cmd_chroot')
+    def test__has_systemd_with_exception(self, mock_exec_cmd_chroot):
+        mock_exec_cmd_chroot.side_effect = Exception()
+
+        result = self.morphing_tools._has_systemd()
+
+        self.assertFalse(result)
+
+    @mock.patch.object(base.BaseLinuxOSMorphingTools, '_exec_cmd_chroot')
+    @mock.patch.object(suse.re, 'search')
```

Comment:
You don't need to mock `re.search`. Use an actual input value for `module` so that:
```python
module_match = re.search("%s.*" % module, available_modules)
```

returns an actual matched value.

For example, use the following in the unit test:
```python
mock_exec_cmd_chroot.return_value = b"module1\nmodule2\nmodule3"

self.morphing_tools._enable_sles_module("module2")
```

And adjust the assertions accordingly.